Client/Clique/Modules/EventDetail/EventDetailViewController.swift

```diff
@@ -20,32 +22,175 @@ class EventDetailViewController: UIViewController {
+        _eventDescription.text = event.description
+        _eventCreator.text = event.creator.name + " " + event.creator.surname
+        _eventLocation.text = event.location
+        _eventTimestamp.text = event.timeStampToString(timestamp: event.timestamp)
+        _eventParticipantsNum.text = String(event.participantNumber)
+        let currency = event.currency
+        if let price = event.cost {
             _eventCost.text = price == 0 ? Constants.Labels.labelFree : String(price) + " " + (currency ?? "")
         } else {
             _eventCost.text = Constants.Labels.labelFree
         }
+        _eventCategory.text = event.category
+        
+        cosmosView.settings.fillMode = .half
+        cosmosView.settings.minTouchRating = 0.5
+        cosmosView.rating = 0
+        
+        if(event.participants.count <= event.participantNumber)
+        {
+            checkUserStatusOnEvent(participants: event.participants)
+        }
+
         
-        _eventCategory.text = _event.category
+    }
+    
+    func didItEnd(timestamp: TimeInterval) -> Event.status {
+        let currentTimestamp = Date().timeIntervalSince1970
+        if(currentTimestamp < timestamp) {
+            return .pending
+        } else if(currentTimestamp > timestamp) {
+            return .done
+        } else {
+            return .inProgress
+        }
+    }
+    
+    func checkUserStatusOnEvent(participants: [Participant?]){
+        var id = 0
+        do {
+            let jwt = try decode (jwt: UserStorage.token!)
+            if let userId = jwt["UserId"].string {
+                id = Int(userId) ?? 0
+            }
+        } catch {
+            return
+        }
+        participants.forEach{participant in
+            if(participant?.user.id  == id) {
+                status = participant?.status ?? 0
+                activateButton()
+                return
+            }
+        }
+    }
+    
+    func activateButton(){
+        eventStatus = didItEnd(timestamp: event.timestamp)
+        if(eventStatus == Event.status.pending){
+            cosmosView.isHidden = true
+            buttonJoinEvent.addTarget(self, action: #selector(registerToEventAlert), for: .touchUpInside)
+            buttonJoinEvent.isHidden = false
+            if(status == 1){
+                buttonJoinEvent.setTitle("Join", for: .normal)
+                alertMessage = Constants.Alerts.joinEventMessage
+            } else if(status == 2) {
+                buttonJoinEvent.setTitle("Leave", for: .normal)
+                alertMessage = Constants.Alerts.cancelEventMessage
+            } else if(status == 3) {
+                buttonJoinEvent.setTitle("You cannot rejoin event", for: .normal)
+                buttonJoinEvent.isEnabled = false
+            }
+        } else if(eventStatus == Event.status.done && status == 2) {
+            cosmosView.isUserInteractionEnabled = false
+            getRatedEvent()
+        } else {
+            buttonJoinEvent.isHidden = true
```

Comment:
Please fix formatting, you do not need to put `if` condition inside brackets. You can just use it like: `if condition == true { ... }`

Fix this everywhere inside project.